In [2]:
# Import pandas
import pandas as pd

In [3]:
# Load the iris dataset from a URL
# df = dataframe = 2-dimensional array of columns and rows
df = pd.read_csv("https://raw.githubusercontent.com/uiuc-cse/data-fa14/gh-pages/data/iris.csv")

In [4]:
df

sepal_length  sepal_width  petal_length  petal_width    species
0             5.1          3.5           1.4          0.2     setosa
1             4.9          3.0           1.4          0.2     setosa
2             4.7          3.2           1.3          0.2     setosa
3             4.6          3.1           1.5          0.2     setosa
4             5.0          3.6           1.4          0.2     setosa
..            ...          ...           ...          ...        ...
145           6.7          3.0           5.2          2.3  virginica
146           6.3          2.5           5.0          1.9  virginica
147           6.5          3.0           5.2          2.0  virginica
148           6.2          3.4           5.4          2.3  virginica
149           5.9          3.0           5.1          1.8  virginica

[150 rows x 5 columns]

# Selecting rows and columns

In [6]:
# Display row 'species' - not recommended to do it like this
# Returns a data series
df['species']

0         setosa
1         setosa
2         setosa
3         setosa
4         setosa
         ...    
145    virginica
146    virginica
147    virginica
148    virginica
149    virginica
Name: species, Length: 150, dtype: object

In [8]:
# Display two columns - again not recommended to do like this
df[['petal_length', 'species']]

petal_length    species
0             1.4     setosa
1             1.4     setosa
2             1.3     setosa
3             1.5     setosa
4             1.4     setosa
..            ...        ...
145           5.2  virginica
146           5.0  virginica
147           5.2  virginica
148           5.4  virginica
149           5.1  virginica

[150 rows x 2 columns]

In [10]:
# Display rows 2 (inclusive) to 6 (exclusive)
df[2:6]

sepal_length  sepal_width  petal_length  petal_width species
2           4.7          3.2           1.3          0.2  setosa
3           4.6          3.1           1.5          0.2  setosa
4           5.0          3.6           1.4          0.2  setosa
5           5.4          3.9           1.7          0.4  setosa

In [12]:
# Display two columns, 4 rows - not recommened again
df[['petal_length', 'species']][2:6]

petal_length species
2           1.3  setosa
3           1.5  setosa
4           1.4  setosa
5           1.7  setosa

Using .loc or .iloc:
.loc uses labels of rows s and columns (for example sepal_length and index row labels)
.iloc uses position of rows and columns ( 0 - 4 for columns, 0 to 149 for rows)

In [15]:
# .loc in this case shows rows with labels 2 - 6
df.loc[2:6]

sepal_length  sepal_width  petal_length  petal_width species
2           4.7          3.2           1.3          0.2  setosa
3           4.6          3.1           1.5          0.2  setosa
4           5.0          3.6           1.4          0.2  setosa
5           5.4          3.9           1.7          0.4  setosa
6           4.6          3.4           1.4          0.3  setosa

In [17]:
# note difference when using .iloc which shows positions 2 to 6 (exclusive)
# even though in this case labels and positions have the same numbering, they are not treated equally
df.iloc[2:6]

sepal_length  sepal_width  petal_length  petal_width species
2           4.7          3.2           1.3          0.2  setosa
3           4.6          3.1           1.5          0.2  setosa
4           5.0          3.6           1.4          0.2  setosa
5           5.4          3.9           1.7          0.4  setosa

In [19]:
# To display a certain column
# : indicates all rows
df.loc[:, 'species']

0         setosa
1         setosa
2         setosa
3         setosa
4         setosa
         ...    
145    virginica
146    virginica
147    virginica
148    virginica
149    virginica
Name: species, Length: 150, dtype: object

In [22]:
df.loc[2:6, ['sepal_length', 'species']]

sepal_length species
2           4.7  setosa
3           4.6  setosa
4           5.0  setosa
5           5.4  setosa
6           4.6  setosa

In [24]:
# Show row with position 2 ( = 3rd row of the dataset)
df.iloc[2]

sepal_length       4.7
sepal_width        3.2
petal_length       1.3
petal_width        0.2
species         setosa
Name: 2, dtype: object

In [25]:
# Show the second column (position 1) and rows with position 2 and 3 ( 2 - 4 exclusive)
df.iloc[2:4, 1]

2    3.2
3    3.1
Name: sepal_width, dtype: float64

In [27]:
# To display a certain single position: .at which goes by label (row with label 3, column with label 'species')
df.at[3, 'species']

'setosa'

In [29]:
#  To display the same using .iat (position)
df.iat[3, 4]

'setosa'

In [31]:
# Show every second row from positions 1 to 10 (exclusive)
df.iloc[1:10:2]

sepal_length  sepal_width  petal_length  petal_width species
1           4.9          3.0           1.4          0.2  setosa
3           4.6          3.1           1.5          0.2  setosa
5           5.4          3.9           1.7          0.4  setosa
7           5.0          3.4           1.5          0.2  setosa
9           4.9          3.1           1.5          0.1  setosa